tensorflow_datasets/core/download/download_manager.py

```diff
@@ -514,25 +515,32 @@ def callback(path):
       return self._extract(resource)
     return self._download(resource).then(callback)
 
-  def download_kaggle_data(self, dataset_name: str):
+  def download_kaggle_data(self, competition: str):
     """Download data for a given Kaggle Dataset or competition.
 
     Note: This function requires Kaggle CLI tool. Read the installation guide
     at https://www.kaggle.com/docs/api.
 
     Args:
-      dataset_name: Dataset name (`zillow/zecon`) or competition name
-        (`titanic`)
+      competition: Dataset name (`zillow/zecon`) or competition name (`titanic`)
 
     Returns:
       The path to the downloaded files.
     """
-    with self._downloader.tqdm():
-      kaggle_downloader = self._downloader.kaggle_downloader(dataset_name)
-      urls = kaggle_downloader.competition_urls
-      files = kaggle_downloader.competition_files
-      return _map_promise(self._download,
-                          dict((f, u) for (f, u) in zip(files, urls)))
+    kaggle_downloader = kaggle.KaggleCompetitionDownloader(competition)
+    download_path = os.path.join(self._download_dir,
+                                 kaggle_downloader.download_dir_name)
+    # If the dataset has already been downloaded, return the path to it.
+    if os.path.isdir(download_path):
+      logging.info('Dataset %s already downloaded: reusing %s.',
+                   competition, download_path)
+      return download_path
```

Comment:
What if the zip file was extracted ? then the output path here won't match where the data was actually extracted.

Reply:
Yes it will. download_path is the path to where the kaggle data is downloaded. If this data is zipped, then it is unzipped.